streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -91,7 +91,7 @@ public static String nameSpaceFromTaskIdAndStore(final String taskIDString, fina
      * @return
      */
     public static String taskIDfromCacheName(final String cacheName) {
-        String[] tokens = cacheName.split("-");
+        String[] tokens = cacheName.split("-", 2);
```

Comment:
This isn't actually a problem, but I did it for consistency with the following change.